packages/app/src/widgets/borrow-flow/components/single-borrow-flow/single-borrow-flow.tsx

```diff
@@ -0,0 +1,68 @@
+'use client'
+
+import React, { useState } from 'react'
+import { SupportedToken } from '@/shared/stellar-constants/tokens'
+import { BorrowStepModal } from '../borrow-step-modal'
+import { CollateralStepModal } from '../collateral-step-modal'
+
+enum Step {
+  Borrow = 'Borrow',
+  Collateral = 'Collateral',
+}
+
+interface Props {
+  type: SupportedToken
+}
+
+const collateralByBorrow: Record<SupportedToken, [SupportedToken, SupportedToken]> = {
+  usdc: ['xlm', 'xrp'],
+  xlm: ['usdc', 'xrp'],
+  xrp: ['usdc', 'xlm'],
+}
```

Comment:
Let's better write something like

```ts
const SUPPORTED_TOKENS = ['xlm', 'xrp', 'usdc'] as const
type SupportedToken = (typeof SUPPORTED_TOKENS)[number]

const getCollateralsByBorrow = <T extends SupportedToken>(
  token: T,
): Array<Exclude<SupportedToken, T>> =>
  SUPPORTED_TOKENS.filter((element) => element !== token) as Array<Exclude<SupportedToken, T>>
```

The Record approach seems to be fragile.

Reply:
Thank you!